runtime/caml/platform.h

```diff
@@ -61,6 +61,11 @@ Caml_inline uintnat atomic_load_relaxed(atomic_uintnat* p)
   return atomic_load_explicit(p, memory_order_relaxed);
 }
 
+Caml_inline value atomic_load_relaxed_val(value* p)
+{
+  return atomic_load_explicit(Op_atomic_val(p), memory_order_relaxed);
+}
+
```

Comment:
Instead of duplicating the function to work with different signatures, would it help to make it a macro so that it is polymorphic? (e.g. as in https://github.com/gadmm/ocaml/compare/custom_thread_safety2..custom_thread_safety)
Would that help with writing the code differently?